client/starwhale/utils/config.py

```diff
@@ -248,3 +257,42 @@ def update_instance(
         )
 
         update_swcli_config(**self._config)
+
+
+@click.group(
+    "config",
+    cls=AliasedGroup,
+    help="Configuration management, edit is supported now",
+)
+def config_cmd() -> None:
+    pass
+
+
+@config_cmd.command("edit", aliases=["e"], help="edit the configuration of swlci")
+def __edit() -> None:
+    _edit()
+
+
+def _edit() -> None:
+    _editor = ""
+    if platform == "linux" or platform == "linux2":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            _editor = os.environ.get("EDITOR", "vi")
+    elif platform == "darwin" or platform == "win32":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            raise NoSupportError(f"No {SWEnv.sw_editor} is configured")
+    else:
+        raise NoSupportError(f"Current platform is not supported by: {platform}")
+    path_config = get_swcli_config_path()
+    fd, fname = tempfile.mkstemp()
+    with open(path_config) as config_f, os.fdopen(fd, "w") as tmpf:
+        for line in config_f:
+            tmpf.write(line)
+        tmpf.close()
+        cmd = _editor + " " + fname
+        _c = subprocess.call(cmd, shell=True)
+        if _c:
+            raise ExistedError("editing file with unexpected failure")
+        os.rename(fname, path_config)
```

Comment:
- It would be better if we support `VISUAL` env  https://unix.stackexchange.com/questions/4859/visual-vs-editor-what-s-the-difference
- `swcli config edit` is not a high-frequency operation, `SWEnv.sw_editor` may not be used by most of the users.
- `VISUAL` and `EDITOR` should be supported in `darwin`
- Using a temp file may not be necessary (users my lose "swap detection" features when using vim)

Refer to the implementation of `pip config edit`
https://github.com/pypa/pip/blob/c6783501f16b7ff8c54adf2bcbc1f5f1d11ac699/src/pip/_internal/commands/configuration.py#L225

Reply:
> * `swcli config edit` is not a high-frequency operation, `SWEnv.sw_editor` may not be used by most of the users.
> Refer to the implementation of pip config edit
https://github.com/pypa/pip/blob/c6783501f16b7ff8c54adf2bcbc1f5f1d11ac699/src/pip/_internal/commands/configuration.py#L225

`pip config edit` offers an option for users to customize the editor, which functions the same like `SWEnv.sw_editor`. Users may not change the editor very frequently. So, is environment variable better than command line option?

Reply:
> * It would be better if we support `VISUAL` env  https://unix.stackexchange.com/questions/4859/visual-vs-editor-what-s-the-difference
> * `swcli config edit` is not a high-frequency operation, `SWEnv.sw_editor` may not be used by most of the users.
> * `VISUAL` and `EDITOR` should be supported in `darwin`
> * Using a temp file may not be necessary (users my lose "swap detection" features when using vim)
> 
> Refer to the implementation of `pip config edit` https://github.com/pypa/pip/blob/c6783501f16b7ff8c54adf2bcbc1f5f1d11ac699/src/pip/_internal/commands/configuration.py#L225

done